src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs

```diff
@@ -411,6 +404,37 @@ public void RenameParameter(string expressionBase, string expectedExpression, st
             }
         }
 
+        [Theory]
+        [InlineData("firstos.option_1 <> Os1Value", "newOsName.Option1 <> Os1Value", "firstos", "newOsName")]        
+        [InlineData("firstos.option_1 <> firstos.option_2", "firstos.option_5 <> firstos.option_2", "firstos.option_1", "option_5")]
+        [InlineData("firstos.option_1 <> Os1Value", "firstos.option_1 <> Os1Value", "Unknown", "Unknown2")]
+        [InlineData("firstos.option_1 <> firstos.option_2", "firstos.option_1 <> firstos.option_2", "firstos.Unknown", "Unknown2")]
+        public void RenameParameter_OptionSet(string expressionBase, string expectedExpression, string oldName, string newName)
+        {
+            PowerFxConfig config = new PowerFxConfig();
+            OptionSet optionSet1 = new OptionSet("firstos", DisplayNameUtility.MakeUnique(new Dictionary<string, string>()
+            {
+                    { "option_1", "Option1" },
+                    { "option_2", "Option2" }
+            }));
+
+            config.AddOptionSet(optionSet1, new DName("TestFirstOptionSet"));
+
+            RecordType r1 = RecordType.Empty()                
+                .Add(new NamedFormulaType("Os1Value", optionSet1.FormulaType, "DisplayOS1Value"));
```

Comment:
Another test: Add a member to this RecordType with logical name `firstos`, make sure that references to that don't get renamed at the same time as references to the global, which would need to be accessed as `[@firstos]` if that collision exists

Reply:
can you add a test demonstrating this scenario works?

Reply:
done